samples/boards/nordic/coresight_stm/README.rst

```diff
@@ -0,0 +1,420 @@
+.. _nrf-coresight-stm-sample:
+
+Coresight STM demo
+##################
+
+Overview
+********
+
+This sample presents how to enable STM logging on nRF54H20 platform.
+
+Also, it prints timing for different log messages.
+Thus, performance of different loggers can be compared.
+
+There are three sample configurations in the :file:`sample.yaml`.
+
+* ``sample.boards.nrf.coresight_stm.local_uart``
+
+This configuration doesn't use STM.
+
+Logs are printed on local console.
+
+* ``sample.boards.nrf.coresight_stm``
+
+This configuration use STM.
+
+Application, Radio, PPR and FLPR cores send logs to an ETR buffer.
+Proxy (Application core) gets logs from the ETR buffer, decodes STPv2 data
+and prints human readable logs on UART.
+
+* ``sample.boards.nrf.coresight_stm.dict``
+
+This sample uses STM logging in dictionary mode.
+
+Application, Radio, PPR and FLPR cores
+send logs to the ETR buffer. Proxy (Application core) forwards data from the ETR to
+the host using UART. Host tool is needed to decode the logs.
+
+Requirements
+************
+
+This application uses nRF54H20 DK board for the demo.
+
+Building and running
+********************
+
+To build the sample, use configuration setups from the :file:`sample.yaml` using the ``-T`` option.
+See the example:
+
+nRF54H20 DK
+
+  .. code-block:: console
+
+     west build -p -b nrf54h20dk/nrf54h20/cpuapp -T sample.boards.nrf.coresight_stm .
+
+Sample Output
+=============
+
+.. code-block:: console
+
+   *** Using Zephyr OS v3.6.99-5bb7bb0af17c ***
+   [00:00:00.227,264] <inf> app/app: test no arguments
+   [00:00:00.227,265] <inf> app/app: test no arguments
+   [00:00:00.227,267] <inf> app/app: test no arguments
+   [00:00:00.227,268] <inf> app/app: test no arguments
+   [00:00:00.227,270] <inf> app/app: test no arguments
+   [00:00:00.227,272] <inf> app/app: test no arguments
+   [00:00:00.227,273] <inf> app/app: test no arguments
+   [00:00:00.227,275] <inf> app/app: test no arguments
+   [00:00:00.227,276] <inf> app/app: test no arguments
```

Comment:
Consider adding `...` to replace intermediate lines and maybe just keep 5 first / 5 last sort of thing

Reply:
I made this section shorter.
I think, this sample shows possibility of printing logs from multiple cores.
Thus, at the beginning, I left log 'test with one argument 100' from each of four cores.
And, whole benchmark result from all cores.